### DISCUSSION TITLE
Gentoo Stage3 Tarballs

### DISCUSSION THREAD

**Discussion by immolo** [frno7/mipsr5900el-gentoo-linux-gnu#7] [Ideas]
Moving here based on our discussion yesterday.

o32 mipsel3 builds should be out in the next 24 hours, I've already thanked Dilfridge and mattst88 on behalf of all of us for listening to everything over the last few days.

I've found everything I need to hack my PS2 now so should be ready to jump in as the stage3 drops although I may need to ask for support with building the kernel as it's always the part I struggle with the most ;)

My plan of action if you agree is; 

1. Verify the glibc build while finding issues and solutions.
2. Create ebuilds for packages we need.
3. Setup a binhost for large files.
4. Update the wiki to reflect the lower barrier to entry for new people to contribute hopefully.

After this then focus on building a MUSL environment to see if the benefits will outweigh the effort it's going to take making that work.

Your thoughts on this @frno7 ?

**Comment by frno7** [comment]
> o32 mipsel3 builds should be out in the next 24 hours, I've already thanked Dilfridge and mattst88 on behalf of all of us for listening to everything over the last few days.

Thanks! But wasn’t it o32 for `mipsel2`, in https://github.com/frno7/linux/issues/71#issuecomment-1110886871?

> I've found everything I need to hack my PS2 now so should be ready to jump in as the stage3 drops although I may need to ask for support with building the kernel as it's always the part I struggle with the most ;)

Good! Compiling the kernel is fairly straight-forward. :-) See also https://github.com/frno7/mipsr5900el-gentoo-linux-gnu/issues/5#issuecomment-1114572591 for some ideas.

> My plan of action if you agree is;
> 
> 1. Verify the glibc build while finding issues and solutions.
> 2. Create ebuilds for packages we need.
> 3. Setup a binhost for large files.
> 4. Update the wiki to reflect the lower barrier to entry for new people to contribute hopefully.

The wiki is important. I’d like to do the mentioned kernel/Gentoo wiki split quite soon. I’m not yet sure how to organise this [wiki](https://github.com/frno7/mipsr5900el-gentoo-linux-gnu/wiki), but articles on how to download and install precompiled archives, and how to configure and build R5900 packages using crossdev, are relevant I think. Most of it should tie in with modern Gentoo practices and tooling, and link to relevant [wiki.gentoo.org](https://wiki.gentoo.org/) articles, in my opinion.

> After this then focus on building a MUSL environment to see if the benefits will outweigh the effort it's going to take making that work.

Musl seems to work, and so I committed 9e1180913b36134d49a299da3646b21b7e663cbc today. See also https://github.com/frno7/mipsr5900el-gentoo-linux-gnu/issues/3#issuecomment-1114653831, and [`Dockerfile`](https://github.com/frno7/mipsr5900el-gentoo-linux-gnu/blob/main/Dockerfile) for the packages now installed with the Docker. It’s an effort-in-progress, of course. To be improved.

**Reply by immolo** [reply]
> Thanks! But wasn’t it o32 for `mipsel2`, in [frno7/linux#71 (comment)](https://github.com/frno7/linux/issues/71#issuecomment-1110886871)?

They are adding a mipsel3 build just for us is the short story after talking it over with them for the last 3 days.

> Good! Compiling the kernel is fairly straight-forward. :-) See also [#5 (comment)](https://github.com/frno7/mipsr5900el-gentoo-linux-gnu/issues/5#issuecomment-1114572591) for some ideas.

It should be easy as I understand the whole process yet for some reason it always fails when I try it on other projects. I don't often have the ear of someone as skilled as you though so I'm hoping I can finally end that curse :)

> The wiki is important. I’d like to do the mentioned kernel/Gentoo wiki split quite soon. I’m not yet sure how to organise this [wiki](https://github.com/frno7/mipsr5900el-gentoo-linux-gnu/wiki), but articles on how to download and install precompiled archives, and how to configure and build R5900 packages using crossdev, are relevant I think. Most of it should tie in with modern Gentoo practices and tooling, and link to relevant [wiki.gentoo.org](https://wiki.gentoo.org/) articles, in my opinion.

The wiki is always the most important part as we are just wasting our time if we don't explain how to use this to others. On the point of the Gentoo Wiki I think we should make a point that everything we add here is also added there too, centralised documentation is what made Gentoo what it is today so we should follow their example to help more people find us. I only found you by pure chance so that will also fix that and solve the issue around it in the kernel bug page.  

> Musl seems to work, and so I committed [9e11809](https://github.com/frno7/mipsr5900el-gentoo-linux-gnu/commit/9e1180913b36134d49a299da3646b21b7e663cbc) today. See also [#3 (comment)](https://github.com/frno7/mipsr5900el-gentoo-linux-gnu/issues/3#issuecomment-1114653831), and [`Dockerfile`](https://github.com/frno7/mipsr5900el-gentoo-linux-gnu/blob/main/Dockerfile) for the packages now installed with the Docker. It’s an effort-in-progress, of course. To be improved.

It will work to boot and you all have done great work already however many packages can need patches written as from speaking to a dev a while back they said no one has been testing it for ages and wanted some help to fix this.
From experience it's easier to use glibc first so you can learn the quirks then when you move to MUSL you have a much better time of knowing if it's a MUSL issue or a system issue.

This is just my suggestion though but we are all working together here so I would like to see what everyone else wants to focus on first as they might have better ideas :)

**Reply by frno7** [reply]
> They are adding a mipsel3 build just for us is the short story after talking it over with them for the last 3 days.

Oh, nice. Will the mipsel2 variant get `-mfix-r5900` too?
